File: algo/45. Jump Game II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    if (nums.length == 1) {
        return 0;
    }
    let max = 0;
    let currEnd = 0;
    let num_jumps = 0;
    for (let i = 0; i < nums.length - 1; i++) {
        max = Math.max(i+nums[i], max);
        if (currEnd == i) {
            num_jumps ++;
            currEnd = max;
        }
    }
    return num_jumps;
};

File: round2/45. Jump Game II.cs
public class Solution {
    public int Jump(int[] nums) {
        if (nums.Length == 1) return 0;

        int currEnd = 0;
        int currMax = 0;
        int numJumps = 0;

        for (int i=0; i<nums.Length-1; i++) {
            currMax = Math.Max(nums[i]+i, currMax);
            if (i == currEnd) {
                numJumps++;
                currEnd = currMax;
            }
        }

        return numJumps;
    }
}